scripts/local/utils/bulk_ops_common.py

```diff
@@ -72,29 +52,79 @@ def get_local_argparser():
     local_argparser.add_argument(
         "-r",
         "--recursive",
-        required=True,
-        type=bool,
+        action='store_true',
         dest="recursive",
         help="Specify whether to process subdirectories recursively",
     )
 
     local_argparser.add_argument(
         "--trigger-size",
         default=200,
-        required=True,
         type=int,
         dest="trigger_size",
-        help="Specify minimum file size to trigger the hook.",
+        help="Specify minimum file size (KB) to trigger the hook.",
     )
+
     return local_argparser
 
 
+def convert_image(input_path, output_path, output_format):
+    """
+    Converts an image to the specified output format.
+    """
+    with Image.open(input_path) as img:
+        if output_format == 'JPG':
+            output_format = 'JPEG'
+        img.save(output_path, output_format)
+
+
+def convert_pdf_to_jpg(input_path, output_dir):
+    """
+    Converts a PDF to a series of JPG images, one per page.
+    """
+    pages = convert_from_path(input_path)
+    for i, page in enumerate(pages):
+        output_path = os.path.join(output_dir, f"page_{i + 1}.jpg")
+        page.save(output_path, 'JPEG')
+
+
+def bulk_convert(input_dir, output_dir, output_format, in_place=False):
+    """
+    Bulk converts images and PDFs to the specified format.
+    """
+    os.makedirs(output_dir, exist_ok=True)
+    extensions = ['png', 'jpg', 'jpeg', 'tiff', 'tif', 'pdf']
+
+    filepaths = walk_and_extract_files(input_dir, extensions)
+
+    for input_path in filepaths:
+        relative_path = os.path.relpath(os.path.dirname(input_path), input_dir)
+        output_subdir = os.path.join(output_dir, relative_path) if not in_place else os.path.dirname(input_path)
+        os.makedirs(output_subdir, exist_ok=True)
+
+        filename = os.path.basename(input_path)
+        if filename.lower().endswith(('.png', '.jpg', '.jpeg', '.tiff', '.tif')):
+            name, _ = os.path.splitext(filename)
+            output_path = os.path.join(output_subdir, f"{name}.{output_format.lower()}")
+            convert_image(input_path, output_path, output_format)
+            print(f"Converted {filename} to {output_format}")
+        elif filename.lower().endswith('.pdf'):
+            pdf_output_dir = os.path.join(output_subdir, os.path.splitext(filename)[0])
+            os.makedirs(pdf_output_dir, exist_ok=True)
+            convert_pdf_to_jpg(input_path, pdf_output_dir)
```

Comment:
Move this logic to a separate script (see the other comment below)
- Apart from making filename as the directory, we should have the filename in the output image as well(suggested in another comment)
- In case of single-page PDFs:
    - A file `image1.pdf` would become `image1/image1_page1.jpg`, but a user may desire to extract the first page into `image1.jpg` (without the directory)
    - let's have a flag in the script that allows this e.g. `--first-page-only`